docs/src/extend/log-management.md

```diff
@@ -0,0 +1,212 @@
+
+Log plugins are installed at `/usr/share/tedge/log-plugins`, by default.
+
+This plugin root directory can be changed using:
+```sh
+sudo tedge config set log.plugin_paths /usr/local/share/tedge/log-plugins,/usr/share/tedge/log-plugins
+```
+
+Multiple plugin directories can be specified if a layered directory structure is desired,
+where plugins from directories earlier in the list gets precedence over the latter ones.
+For example, with the above configuration, a `file` plugin in `/usr/local/share/tedge/log-plugins`
+would override the one with the same name in `/usr/share/tedge/log-plugins`.
+This mechanism can be used to override the factory plugins when needed.
+
+## Permissions
+
+Plugins are executed by the agent with `sudo` privileges.
+The agent automatically creates the following sudoers entry,
+giving sudo rights to all plugins installed at `/usr/local/lib/tedge/log-plugins`:
+
+```
+tedge    ALL = (ALL) NOPASSWD:SETENV: /usr/local/lib/tedge/log-plugins/[a-zA-Z0-9]*
+```
+
+If the `log.plugin_paths` config is updated with additional directories as shown in the previous section,
+then sudoers entries must be created for those directories as well.
+
+Additionally, ensure your plugin has appropriate permissions to access the log sources it needs.
+
+## Creating a Custom Log Plugin
+
+### Example: docker plugin
+
+Here's a `docker` plugin example that can retrieve logs from containers using the `docker logs` command:
+
+```sh
+#!/bin/sh
+set -eu
+
+help() {
+    cat <<EOT
+docker log plugin to retrieve the logs from containers using the docker cli
+
+$0 <SUBCOMMAND>
+
+SUBCOMMANDS
+  list
+  get <type> [--since <timestamp>] [--until <timestamp>]
+EOT
+}
+
+list_log_types() {
+    docker ps -a --format "{{.Names}}"
+}
+
+get_log_by_type() {
+    log_type="$1"
+    shift
+
+    # Parse option defaults
+    since="24h"
+    until="0s"
+
+    while [ $# -gt 0 ]; do
+        case "$1" in
+            --since)
+                since="$2"
+                shift
+                ;;
+            --until)
+                until="$2"
+                shift
+                ;;
+        esac
+        shift
+    done
+
+    # Retrieve logs using docker logs
+    docker logs "$log_type" \
+        --since "$since" \
+        --until "$until"
+}
+
+if [ $# -lt 1 ]; then
+    echo "Missing required subcommand" >&2
+    help
+    exit 1
+fi
+
+SUBCOMMAND="$1"
+shift
+
+case "$SUBCOMMAND" in
+    list)
+        list_log_types
+        ;;
+    get)
+        get_log_by_type "$@"
+        ;;
+    *)
+        echo "Unsupported command" >&2
+        exit 1
+        ;;
+esac
+```
+
+* The `list` command of this plugin will output all container names:
+  ```
+  nginx
+  mosquitto
+  ```
+
+* The `get` command retrieves the logs for the target container using the `docker logs` command.
+
+### Installation
+
+1. Copy the plugin to the plugins directory:
+
+```sh
```

Comment:
The code blocks (for step 1 and 2) need to be indented as they are under a numbered list.

Reply:
resolved by putting it into one line instead (which is also fine by me).